texar/torch/data/tokenizers/bert_tokenizer.py

```diff
@@ -166,7 +166,8 @@ def __init__(self,
         self.wordpiece_tokenizer = WordpieceTokenizer(vocab=self.vocab,
                                                       unk_token=self.unk_token)
 
-    def _map_text_to_token(self, text: str) -> List[str]:  # type: ignore
+    def _map_text_to_token(self, text: str) -> \
+        Union[List[str], List[Tuple[str, int, int]]]:  # type: ignore
         split_tokens = []
```

Comment:
To avoid the typing issue here, you can create two different lists inside the `if` and `else` block:
`split_tokens: List[str] = []` and `split_token_span: List[Tuple[str, int, int]] = []`, and return then separately

Reply:
I think _map_text_to_token should only return split_tokens as List[str]. It doesn't need span info. My thought is that it is a within class function and called by "map_text_to_token", then "map_token_to_id" calls "map_text_to_token". These functions don't need span info.

Reply:
Perhaps separate it out as another function "tokenize_with_span" is better.